utils/errors/errors.go

```diff
@@ -37,13 +37,33 @@ type NotifyExt interface {
 	Notified(status bool)
 }
 
+// RawExt is the interface definition for raw data about an error
+type RawExt interface {
+	RawData() []interface{}
+}
+
 type customError struct {
 	Msg          string
 	stack        []uintptr
 	frame        []StackFrame
 	cause        error
 	shouldNotify bool
 	status       *grpcstatus.Status
+	rawData      []interface{}
+}
+
+type Option func(*customError)
```

Comment:
I'm mostly thinking if it's good to introduce such good extensibility to this low-level package and does this package needed? error package should be simple and provide basic functionalities to others who want to build up more on top of it, say this pattern is good for the customized struct which built on top of Orion error

what's your thought regarding this?